### PR TITLE
Fix/custom classes

### PR DESCRIPTION
**Pull request by ericawright**
Status: **Ready for Review**
Owner: @ericawright
Reviewers: @mikenikles @fractaltheory @jansepar @marlowpayne @stewartyu
## Changes
- user was unable to add custom classes for appendTo and container options (unless they also used pikabu's built in classes)
- $viewport was not being defined when custom classes used
- $animators array was missing custom class when .pikabu_container class was not there
## Todos:
- [ ] Engineer +1
### Feedback:

_none so far_
## How to Test
- in `pikabu/examples/assets/js/main`, add some custom options such as:
  `container: '.m-pikabu-container',
          appendTo: '.m-pikabu-viewport',`
  to the pikabu declarations (make sure to do this to all 4, or else it will look for the default elements which we will remove)
- in `pikabu/examples/index.html` change line 18 `<div class="pikabu">` remove "pikabu" and add custom class name: ".m-pikabu-viewport"
- also in `pikabu/examples/index.html` change line 25 `<div class="pikabu__container">` remove "pikabu_container" and replace with custom class "m-pikabu-container"
- run `grunt serve`
- go to http://localhost:3000/examples/

### CODE REVIEW

**Comment by stewartyu** [comment]
:clap:

File: dist/pikabu.js [hunk 1]
```diff
@@ -131,6 +131,10 @@
             this.$body = $('body');
             this.$animators = $('.' + classes.CONTAINER + ', ' + '.' + classes.FIXED);
 
+            if (this.options.container){
```
Comment (mikenikles) [inline dist/pikabu.js]: @ericawright Could you please add a space between `)` and `{`? This is documented in the Mobify Style Guide, specifically at https://github.com/mobify/mobify-code-style/tree/develop/javascript#use-a-space-before-the-opening-curly-brace-for-function-calls

File: src/js/pikabu.js [hunk 2]
```diff
@@ -131,6 +131,10 @@
             this.$body = $('body');
             this.$animators = $('.' + classes.CONTAINER + ', ' + '.' + classes.FIXED);
 
+            if (this.options.container){
```
Comment (mikenikles) [inline src/js/pikabu.js]: @ericawright Could you please add a space between `)` and `{`? This is documented in the Mobify Style Guide, specifically at https://github.com/mobify/mobify-code-style/tree/develop/javascript#use-a-space-before-the-opening-curly-brace-for-function-calls

File: src/js/pikabu.js [hunk 3]
```diff
@@ -131,6 +131,10 @@
...
+                this.$animators.push($(this.options.container)[0]);
+            };
```
Comment (mikenikles) [inline src/js/pikabu.js]: This semicolon is not required and linters will throw an error. Do you mind removing it?

**Comment by stewartyu** [comment]
Oh yeah... we should have tests to make sure this doesn't regress!

File: dist/pikabu.js [hunk 4]
```diff
@@ -242,7 +246,7 @@
                     }
                 });
 
-            this.$viewport = $('.' + classes.VIEWPORT);
+            this.$viewport = this.options.appendTo ? $(this.options.appendTo) : $('.' + classes.VIEWPORT)
```
Comment (mikenikles) [inline dist/pikabu.js]: Also, there's a semicolon missing at the end of this line according to my editor's linter :wink:

File: src/js/pikabu.js [hunk 5]
```diff
@@ -242,7 +246,7 @@
                     }
                 });
 
-            this.$viewport = $('.' + classes.VIEWPORT);
+            this.$viewport = this.options.appendTo ? $(this.options.appendTo) : $('.' + classes.VIEWPORT)
```
Comment (mikenikles) [inline src/js/pikabu.js]: Also, there's a semicolon missing at the end of this line according to my editor's linter :wink:

**Comment by ericawright** [comment]
accidentally aimed at master

**Pull request closed without merging**